### PR TITLE
Refactor low memory test into generic fault injection

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

To make it easier to implement #2101, this PR refactors and simplifies the fault injection logic. It exposes a simple set of APIs that modules can use to perform deterministic fault injection.

## Testing

CI/CD. Existing low memory tests.

## Documentation

No.

### CODE REVIEW

**Comment by Alan-Jowett** [comment]
@dthaler this is blocked on "low_memory" tests, which I ended up renaming to "fault_injection". Can we get the required tests switched over / suspended until this is merged?

**Comment by dthaler** [comment]
> @dthaler this is blocked on "low_memory" tests, which I ended up renaming to "fault_injection". Can we get the required tests switched over / suspended until this is merged?

Done

File: libs/platform/user/ebpf_fault_injection.h
Comment (dthaler) [inline libs/platform/user/ebpf_fault_injection.h:19]:
```suggestion
     * @retval EBPF_SUCCESS The operation was successful.
     * @retval EBPF_NO_MEMORY Memory allocation failed.
```

File: libs/platform/user/ebpf_fault_injection.h
Comment (dthaler) [inline libs/platform/user/ebpf_fault_injection.h:35]:
```suggestion
     * @retval true Fault should be injected.
     * @retval false Fault should not be injected.
```

File: libs/platform/user/ebpf_fault_injection.cpp
Comment (dthaler) [inline libs/platform/user/ebpf_fault_injection.cpp:33]:
```suggestion
```

File: libs/platform/user/ebpf_fault_injection.cpp
Comment (dthaler) [inline libs/platform/user/ebpf_fault_injection.cpp:149]:
```suggestion
     * @retval true The current thread is recursing from the fault injection callback.
     * @retval false The current thread is not recursing from the fault injection callback.
```

File: libs/platform/user/ebpf_fault_injection.cpp
Comment (dthaler) [inline libs/platform/user/ebpf_fault_injection.cpp:198]:
```suggestion
        // Form the canonical stack.
```

**Review by shpalani — APPROVED** [review]

**Review by gtrevi — APPROVED** [review]

**Pull request merged**
